### PR TITLE
postgresql.pkgs.citus: init at 10.0.3

### PR DESCRIPTION
**Pull request by mkg20001**
###### Motivation for this change

Add citus db

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/sql/postgresql/ext/citus.nix [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+stdenv.mkDerivation rec {
+  pname = "citus";
+  version = "10.0.3";
+
+  src = fetchFromGitHub {
+    owner = "citusdata";
+    repo = "citus";
+    rev = "v${version}";
+    sha256 = "hzR4tQnws5kYyLY8xzoIiOEJrCQFj0t2d+lJg2c8/AE=";
+    fetchSubmodules = false;
```
Comment (SuperSandro2000) [inline pkgs/servers/sql/postgresql/ext/citus.nix:20]:
```suggestion
```
that is the default.

File: pkgs/servers/sql/postgresql/ext/citus.nix [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  nativeBuildInputs = [
+    autoreconfHook
+    zstd.dev
+  ];
+
+  # dirty hack to give us a clean install tree
+  installPhase = ''
+    make install DESTDIR=/tmp/citus
+    mkdir -p $out
+    mv /tmp/citus/nix/store/*/* $out
```
Comment (SuperSandro2000) [inline pkgs/servers/sql/postgresql/ext/citus.nix:39]: This looks really hacky. Also this probably breaks when building two times the package in parallel. It must at least use $TMP or mktemp -d.

File: pkgs/servers/sql/postgresql/ext/citus.nix (hunk 1 above)
Comment (mkg20001) [reply pkgs/servers/sql/postgresql/ext/citus.nix:20]: This is from `nix-prefetch-github owner repo --no-fetch-submodules --nix`, this is a bug there

File: pkgs/servers/sql/postgresql/ext/citus.nix (hunk 2 above)
Comment (mkg20001) [reply pkgs/servers/sql/postgresql/ext/citus.nix:39]: Another way is to change the postgres pkgconfig paths, but I don't know any way to override those.

Comment (SuperSandro2000) [reply pkgs/servers/sql/postgresql/ext/citus.nix:39]: substituteInPlace maybe?

Comment (mkg20001) [reply pkgs/servers/sql/postgresql/ext/citus.nix:39]: The problem is that it's referencing some makefile from postgresql itself (think basically like building a kernel module).

Comment (marsam) [reply pkgs/servers/sql/postgresql/ext/citus.nix:39]: I think you can use the expression of the previous attempt https://github.com/NixOS/nixpkgs/pull/38698/commits/4119c9fde040aa0a938974950ad65318bfa4bf16

**Pull request closed without merging**